src/main/java/com/google/opensesame/user/UserEntity.java

```diff
@@ -0,0 +1,123 @@
+  public ArrayList<String> interestTags;
+  /** Datastore IDs of projects the given user has decided to mentor for. */
+  public ArrayList<String> projectIDs = new ArrayList<String>();
+  /** Datastore IDs of mentees that have expressed interest in being mentored by this user. */
+  public ArrayList<String> menteeIDs = new ArrayList<String>();
+
+  protected UserEntity() {}
+
+  /**
+   * Create a User entity with information from a UserData Instance
+   *
+   * @param user
+   */
+  public UserEntity(UserData userData) {
+    this.gitHubId = userData.getGitHubID();
+    this.interestTags = userData.getTags();
+    this.email = userData.getEmail();
+    this.userId = userData.getUserID();
+    this.projectIDs = userData.getProjectIDs();
+    this.menteeIDs = userData.getMenteeIDs();
+  }
+
+  /**
+   * Create a User entity with the given information. Use this when instantiating a User that is not
+   * a mentor for any project.
+   *
+   * @param userId
+   * @param gitHubId
+   * @param interestTags
+   * @param email
+   */
+  public UserEntity(String userId, String gitHubId, ArrayList<String> interestTags, String email) {
+    this.userId = userId;
+    this.gitHubId = gitHubId;
+    this.interestTags = interestTags;
+    this.email = email;
+  }
+  /**
+   * Create a user with the given information. Use this to instantiate a User that is a mentor
+   * without any mentees.
+   *
+   * @param userId
+   * @param gitHubId
+   * @param interestTags
+   * @param email
+   * @param projectIDs
+   */
+  public UserEntity(
+      String userId,
+      String gitHubId,
+      ArrayList<String> interestTags,
+      String email,
+      ArrayList<String> projectIDs) {
+    this(userId, gitHubId, interestTags, email);
+    this.projectIDs = projectIDs;
+  }
+
+  /**
+   * Create a User entity with the given information.
+   *
+   * @param userId
+   * @param gitHubId
+   * @param interestTags
+   * @param email
+   * @param projectIDs
+   * @param menteeIDs
+   */
+  public UserEntity(
+      String userId,
+      String gitHubId,
+      ArrayList<String> interestTags,
+      String email,
+      ArrayList<String> projectIDs,
+      ArrayList<String> menteeIDs) {
+    this(userId, gitHubId, interestTags, email, projectIDs);
+    this.menteeIDs = menteeIDs;
+  }
+
+  @OnSave
+  protected void setIsMentor() {
+    this.isMentor = this.isMentor();
+  }
+
+  /** true if the entity represents a mentor, false otherwise. */
+  public Boolean isMentor() {
+    return !projectIDs.isEmpty();
+  }
+
+  /**
+   * Add a project Id to the list of project Ids associated with this instance of the mentor object.
+   * Remember to store this change in datastore with a personEntity.
+   */
+  public void addProject(String projectID) {
+    projectIDs.add(projectID);
+  }
+
+  /**
+   * Add a mentee Id to the list of mentee Ids associated with this instance of the mentor object.
+   * Remember to store this change in datastore with a personEntity.
+   */
```

Comment:
Another possible fix.
```suggestion
  /**
   * Add a mentee Id to the list of mentee Ids associated with this user.
   * Remember to store this change by saving the entity back to datastore.
   */
```